src/Plugin/Field/FieldWidget/Select2Widget.php

```diff
@@ -82,4 +83,34 @@ public static function validateElement(array $element, FormStateInterface $form_
     $form_state->setValueForElement($element, $items);
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  protected function getOptions(FieldableEntityInterface $entity) {
+    $options = parent::getOptions($entity);
+    $default_value = $entity->get($this->fieldDefinition->getName())->getValue();
+
+    $order = [];
+    foreach ($default_value as $value) {
+      $order[] = $value['target_id'];
+    }
+
+    // Move selected values to the beginning.
+    uksort($options, function ($a, $b) use ($order, $options) {
+      $a_in_array = in_array($a, $order);
+      $b_in_array = in_array($b, $order);
+
+      if (!$a_in_array && !$b_in_array) {
+        return strcmp($options[$a], $options[$b]);
```

Comment:
Option lists are not always ordered in alphabetical order, this reorders all lists to be alphabetical